pkg/api/helpers/errors.go

```diff
@@ -108,25 +108,29 @@ func ResponseAsError(response interface{}) error {
 			Err:     ErrRequestFailed,
 		}
 	}
-
-	f := r.FieldByName("HTTPResponse")
-	if !f.IsValid() {
-		return fmt.Errorf("[no HTTPResponse]: %w", ErrRequestFailed)
-	}
-	httpResponse, ok := f.Interface().(*http.Response)
-	if !ok {
-		return fmt.Errorf("%w: no HTTPResponse", ErrRequestFailed)
+	var f reflect.Value
+	var httpResponse *http.Response
+	var ok bool
+	if reflect.TypeOf(response).String() != "*http.Response" {
+		f = r.FieldByName("HTTPResponse")
+		if !f.IsValid() {
+			return fmt.Errorf("[no HTTPResponse]: %w", ErrRequestFailed)
+		}
+		httpResponse, ok = f.Interface().(*http.Response)
+		if !ok {
+			return fmt.Errorf("%w: no HTTPResponse", ErrRequestFailed)
+		}
+	} else {
+		httpResponse = response.(*http.Response)
 	}
```

Comment:
As an *alternative* to the above suggestion: even if you want to keep the same function for both cases, you don't need reflection here.  It's  just a checked cast, e.g.:

```suggestion
	if httpResponse, ok = response.(*http.Response); !ok {
		f = r.FieldByName("HTTPResponse")
		if !f.IsValid() {
			return fmt.Errorf("[no HTTPResponse]: %w", ErrRequestFailed)
		}
		httpResponse, ok = f.Interface().(*http.Response)
		if !ok {
			return fmt.Errorf("%w: no HTTPResponse", ErrRequestFailed)
		}
	}
```